.circleci/config.yml

```diff
@@ -1,5 +1,166 @@
-version: 2
+version: 2.1
+
+references:
+  GO_VERSION: &GOVERSION 1.12.1
```

Comment:
probably this needs to be in sync with Dockerfile.ci and other places.  I don't think we handle it correctly in travis now either.